test/helpers/register-setup.ts

```diff
@@ -165,19 +167,25 @@ export const registrationWithSetup = async ({
   //  optimize for the best UX!
   //  maybe add API to SubReg to set these up in one tx?
   // set up prices
-  if (fullConfig.distrConfig.pricerContract === zns.fixedPricer.address) {
+  if (fullConfig.distrConfig.pricerContract === zns.fixedPricer.address && setConfigs) {
     await zns.fixedPricer.connect(user).setPriceConfig(
       domainHash,
-      fullConfig.priceConfig as IFixedPriceConfig,
+      {
+        ...fullConfig.priceConfig as IFixedPriceConfig,
+        isSet: true,
+      },
     );
-  } else if (fullConfig.distrConfig.pricerContract === zns.curvePricer.address) {
+  } else if (fullConfig.distrConfig.pricerContract === zns.curvePricer.address && setConfigs) {
     await zns.curvePricer.connect(user).setPriceConfig(
       domainHash,
-      fullConfig.priceConfig as ICurvePriceConfig,
+      {
+        ...fullConfig.priceConfig as ICurvePriceConfig,
+        isSet: true,
+      },
     );
   }
 
-  if (fullConfig.paymentConfig.beneficiary !== AddressZero) {
+  if (fullConfig.paymentConfig.beneficiary !== AddressZero && setConfigs) {
```

Comment:
mayeb simplify this and the above `if` statements that all rely on `setConfigs` by having one `if(!setConfigs) throw ...` at the top